pysph/tools/interpolator.py

```diff
@@ -27,12 +30,113 @@ def post_loop(self, d_idx, d_prop, d_number_density):
             d_prop[d_idx] /= d_number_density[d_idx]
 
 
-class InterpolateSPH(Equation):
-    def initialize(self, d_idx, d_prop):
-        d_prop[d_idx] = 0.0
+class SPHFirstOrderApproximationPreStep(Equation):
+    def __init__(self, dest, sources, dim=1):
+        self.dim = dim
+
+        super(SPHFirstOrderApproximationPreStep, self).__init__(dest, sources)
+
+    def initialize(self, d_idx, d_mommat):
+        i, j = declare('int', 2)
+
+        for i in range(4):
+            for j in range(4):
+                d_mommat[16*d_idx + j+4*i] = 0.0
```

Comment:
Instead of `mommat`, can't it just be `moments`?

Reply:
done